### PR TITLE
feat: add C implementation for `@stdlib/stats/base/dists/bernoulli/mean`

### LINKED ISSUE #3386: [RFC]: Add C implementation for `@stdlib/stats/base/dists/arcsine/logcdf`
### Description

This RFC proposes adding a C implementation, including native Node.js addon, C benchmarks and C examples, for the `@stdlib/stats/base/dists/arcsine/logcdf` package.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/arcsine/logcdf" make install-node-addons
```

#### Run C Examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/logcdf/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/logcdf/benchmark/c/benchmark.c"
```

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### LINKED ISSUE #3414: [RFC]: Add C implementation for `@stdlib/stats/base/dists/bernoulli/mean`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/bernoulli/mean`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/bernoulli/mean" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/bernoulli/mean/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by vivekmaurya001**
Resolves #3414.

## Description

- adds C implementation for @stdlib/stats/base/dists/bernoulli/mean along with relevant tests, benchmarks and examples

This pull request:

- Progresses: https://github.com/stdlib-js/stdlib/issues/3414
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
Until CI passes, this PR cannot be reviewed.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/README.md [hunk 1]
```diff
@@ -130,6 +130,92 @@ for ( i = 0; i < 10; i++ ) {
...
+Returns the [expected value][expected-value] of a [Bernoulli][bernoulli-distribution] distribution with success probability `p`.
+
+```c
+double out = stdlib_base_dists_bernoulli_mean( 0.1 );
+// returns ~0.1
+```
+
+The function accepts the following arguments:
+
+-   **p**: `[in] double` success probability
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/README.md]:
```suggestion
-   **p**: `[in] double` success probability.
```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/README.md [hunk 2]
```diff
@@ -130,6 +130,92 @@ for ( i = 0; i < 10; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/stats/base/dists/bernoulli/mean.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main( void ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/README.md:200]:
```suggestion

int main( void ) {
```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+var mean = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( mean instanceof Error )
+};
+
+
+// MAIN //
+
+bench( pkg, opts, function benchmark( b ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/benchmark.native.js]:
```suggestion
bench( pkg+'::native', opts, function benchmark( b ) {
```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/c/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+*/
+static double benchmark( void ) {
+	double elapsed;
+	double p[ 100 ];
+	double y;
+	double t;
+	int i;
+
+    for ( i = 0; i < 100; i++ ) {
+		p[ i ] = (double)r / ( (double)RAND_MAX + 1.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/c/benchmark.c]:
```suggestion
		p[ i ] = rand_double();
```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/c/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::%s\n", NAME );
+		elapsed = benchmark();
+		print_results( elapsed );
+		printf( "ok %d benchmark finished\n", i+1 );
+	}
+	print_summary( REPEATS, REPEATS );
+}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/c/benchmark.c]: Trailing newlines.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/include/stdlib/stats/base/dists/bernoulli/mean.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Returns the mean of a Bernoulli distribution.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/include/stdlib/stats/base/dists/bernoulli/mean.h]:
```suggestion
* Returns the mean of a Bernoulli distribution with success probability `p`.
```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/lib/native.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Returns the expected value of a Bernoulli distribution.
+*
+* @param {Probability} p - success probability
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/lib/native.js:32]:
```suggestion
* @private
* @param {Probability} p - success probability
```

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/manifest.json [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+          "./include"
+        ],
+        "libraries": [],
+        "libpath": [],
+        "dependencies": [
+          "@stdlib/math/base/assert/is-nan"
+        ]
+      }
+    ]
+  }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/manifest.json]: Indentation and trailing newline.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/src/main.c [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+* Returns the expected value of a Bernoulli distribution.
+*
+* @param p    success probability
+* @return     expected value
+*
+* @example
+* double y = stdlib_base_dists_bernoulli_mean( 0.1 );
+* // returns ~0.1
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/src/main.c]: Why are these return values indicated as approximate values?

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/README.md [hunk 10]
```diff
@@ -130,6 +130,92 @@ for ( i = 0; i < 10; i++ ) {
...
+#include "stdlib/stats/base/dists/bernoulli/mean.h"
+```
+
+#### stdlib_base_dists_bernoulli_mean( p )
+
+Returns the [expected value][expected-value] of a [Bernoulli][bernoulli-distribution] distribution with success probability `p`.
+
+```c
+double out = stdlib_base_dists_bernoulli_mean( 0.1 );
+// returns ~0.1
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/README.md]: Why is this value displayed as an approximate value?

**Comment by kgryte** [comment]
You'll want to go through your other related PRs to make any similar requested changes.

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/src/main.c [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+*
+* @example
+* double y = stdlib_base_dists_bernoulli_mean( 0.5 );
+* // returns ~0.5
+*/
+double stdlib_base_dists_bernoulli_mean( const double p ) {
+	double q;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/src/main.c]: This value is not used; hence, the C lint failure.

**Comment by vivekmaurya001** [comment]
@kgryte , done changes please give a look

File: lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/c/benchmark.c [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+* @return elapsed time in seconds
+*/
+static double benchmark( void ) {
+	double elapsed;
+	double p[ 100 ];
+	double y;
+	double t;
+	int i;
+
+    for ( i = 0; i < 100; i++ ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/bernoulli/mean/benchmark/c/benchmark.c]:
```suggestion
	for ( i = 0; i < 100; i++ ) {
```

**Review by kgryte — APPROVED** [review]
Thanks for working on this @vivekmaurya001. This now LGTM.

**Pull request merged**